### DISCUSSION TITLE
Is it possible to use update (or replace with index) with only partial attributes?

### DISCUSSION THREAD

**Discussion by arnab** [isar/isar#1308] [Help]
Looking to see if a Firebase style partial update (see [Read and Write Data  |  Firebase Realtime Database](https://firebase.google.com/docs/database/flutter/read-and-write#updating_or_deleting_data)) is possible with Isar. 

I have an object I am storing in Isar with ~10 fields in it. In some circumstances, I need to update 1-2 properties on the object.

What I would like to do is with either:
* with [Isar update](https://isar.dev/crud.html#modifying-the-database)
* or with [Isar Replace/Put Index](https://isar.dev/indexes.html#replace-indexes)

... update only a few properties of an existing object (without needing to read it first, then changing a few properties and then calling `put` again).

Is this possible with Isar?

**Comment by arnab** [comment]
Looks like it's not supported (since there is no response here).

Opened as a feature request: [Support (Firebase style) partial update to objects stored in Isar · Issue #1309 · isar/isar](https://github.com/isar/isar/issues/1309).

**Reply by simc** [reply]
Thanks! I'm already working on it for v4
